iceoryx_dds/source/iceoryx_dds/dds/cyclone_data_reader.cpp

```diff
@@ -53,25 +55,62 @@ void iox::dds::CycloneDataReader::connect() noexcept
     }
 }
 
-iox::cxx::optional<uint32_t> iox::dds::CycloneDataReader::peekNextSize()
+iox::cxx::optional<iox::dds::IoxChunkDatagramHeader> iox::dds::CycloneDataReader::peekNextIoxChunkDatagramHeader()
 {
     // ensure to only read sample - do not take
-    auto readSamples = m_impl.select().max_samples(1u).state(::dds::sub::status::SampleState::any()).read();
+    auto readSamples = m_impl.select().max_samples(1U).state(::dds::sub::status::SampleState::any()).read();
 
-    if (readSamples.length() > 0)
+    constexpr iox::cxx::nullopt_t NO_VALID_SAMPLE_AVAILABLE;
+
+    if (readSamples.length() == 0)
+    {
+        return NO_VALID_SAMPLE_AVAILABLE;
+    }
+
+    auto nextSample = readSamples.begin();
+    auto& nextSamplePayload = nextSample->data().payload();
+    auto nextSampleSize = nextSamplePayload.size();
+
+    // Ignore samples with no payload
+    if (nextSampleSize == 0)
     {
-        auto nextSample = readSamples.begin();
-        auto nextSampleSize = nextSample->data().payload().size();
+        return NO_VALID_SAMPLE_AVAILABLE;
+    }
 
-        // Ignore samples with no payload
-        if (nextSampleSize != 0)
+    // Ignore Invalid IoxChunkDatagramHeader
+    if (nextSampleSize < sizeof(iox::dds::IoxChunkDatagramHeader))
+    {
+        auto log = LogError();
+        log << "[CycloneDataReader] invalid sample size! Must be at least sizeof(IoxChunkDatagramHeader) = "
+            << sizeof(iox::dds::IoxChunkDatagramHeader) << " but got " << nextSampleSize;
+        if (nextSampleSize >= 1)
         {
-            return iox::cxx::optional<uint32_t>(static_cast<uint32_t>(nextSampleSize));
+            log << "! Potential datagram version is " << static_cast<uint16_t>(nextSamplePayload[0])
+                << "! Dropped sample!";
         }
+        return NO_VALID_SAMPLE_AVAILABLE;
+    }
+
+    if (nextSamplePayload[0] != iox::dds::IoxChunkDatagramHeader::DATAGRAM_VERSION)
+    {
+        LogError()
```

Comment:
Could you please also print the required `DATA_GRAM_VERSION` and the version we received. Could be helpful for debugging.